### PR TITLE
Support FLOAT32 type in Spanner Migration Tool.

### PR DESCRIPTION
**Pull request by arawind**
This changes the existing behavior of widening a source float32 type into float64 in spanner, and instead mapping the source float32 type to spanner's new float32 type.

This is a re-patch of https://github.com/GoogleCloudPlatform/spanner-migration-tool/pull/785

- [x] Tests pass
- [x] Appropriate changes to README are included in PR

### CODE REVIEW

**Comment by manitgupta** [comment]
Actually this made me wonder if the changes around Float32 - Dataflow and SMT, are backward compatible? 

If I have a customer who has already mapped float to float64 in a previous schema migration, and uses the new dataflow template with the changes you made recently, will anything break?

Please walk through the cases therein once. (`Schema version` X `Dataflow template version`)

**Comment by arawind** [comment]
@manitgupta IIUC if the spanner migration tool has already converted a schema and float32 from sourcedb is mapped to float64, DataFlow will still process the events correctly as float64 values.

I believe this happens because DataFlow templates use the current Spanner schema to make the choice of whether to parse the input change as FLOAT64 or FLOAT32. 

So these steps should result in a backward compatible migration. LMK if I missed something!

**Comment by manitgupta** [comment]
Yes, you're right. The spanner schema is used to infer the data type. So last as long the spanner schema has something that the pipeline is able to cast to, we will be good.

On the other hand, if the customer decides to regenerate the spanner schema using SMT, there data types will change to Float32 from Float64. This should be okay if the customer is redoing a migration.

This leaves only the `load from session file scenario` which I think is worth validating - 

Consider this example - 

1. Customer did a schema migration and generated a session file.
2. Customer wants to rename a column now from A -> B.
3. Customer loads the session file into SMT.
4. Customer makes the change to the column name.

Can you validate what is the value of spanner data type that appears from when loading from session file? Ideally, it should load from `spSchema` and hence the newly changed defaults should not matter.

**Comment by arawind** [comment]
Thanks @manitgupta!

As mentioned offline, I've verified that the loading the old session file still uses the previous mapping of `FLOAT32 source => FLOAT64 in spanner` and does not change it to FLOAT32 in spanner.

**Review by shreyakhajanchi — COMMENTED** [review]
Just one comment regarding float -> FLOAT64 mapping option

File: sources/mysql/toddl.go [hunk 1]
```diff
@@ -113,7 +113,7 @@ func toSpannerTypeInternal(srcType schema.Type, spType string) (ddl.Type, []inte
 		case ddl.String:
 			return ddl.Type{Name: ddl.String, Len: ddl.MaxLength}, []internal.SchemaIssue{internal.Widened}
 		default:
```
[the remark this answers is not in the document]
Comment (arawind) [reply sources/mysql/toddl.go:117]: Makes sense, thanks! Done.

**Comment by arawind** [comment]
@manitgupta @shreyakhajanchi gentle ping!

**Review by shreyakhajanchi — APPROVED** [review]
LGTM

**Pull request merged**